spec/helpers/pageflow/files_helper_spec.rb

```diff
@@ -43,5 +43,122 @@ module Pageflow
         expect(helper.file_duration(audio_file)).to eq('-')
       end
     end
+
+    describe '#files_json_seeds' do
+      it 'has keys of all file types' do
+        revision = create(:revision, :published)
+        entry = create(:entry, published_revision: revision)
+        published_entry = PublishedEntry.new(entry)
+
+        files_seed = JSON.parse(helper.files_json_seeds(published_entry))
+
+        expect(files_seed).to have_key('image_files')
+        expect(files_seed).to have_key('video_files')
+        expect(files_seed).to have_key('audio_files')
+      end
+
+      it 'seeds required data for an image file' do
+        revision = create(:revision, :published)
+        entry = create(:entry, published_revision: revision)
+        published_entry = PublishedEntry.new(entry)
+        image_file = create(:image_file)
+        create(:file_usage, revision: revision, file: image_file)
+
+        files_seed = JSON.parse(helper.files_json_seeds(published_entry))
+        image_file_seed = files_seed['image_files'].first
+
+        expect(image_file_seed).to have_key('id')
+        expect(image_file_seed).to have_key('state')
+        expect(image_file_seed).to have_key('rights')
+        expect(image_file_seed).to have_key('usage_id')
+        expect(image_file_seed).to have_key('retryable')
+        expect(image_file_seed).to have_key('file_name')
+        expect(image_file_seed).to have_key('width')
+        expect(image_file_seed).to have_key('height')
+        expect(image_file_seed).to have_key('panorama_url')
+        expect(image_file_seed).to have_key('dimensions')
+        expect(image_file_seed).to have_key('thumbnail_url')
+        expect(image_file_seed).to have_key('link_thumbnail_url')
```

Comment:
I'm not convinced reiterating all of the attributes provides value here. From the perspective of the functionality we are testing, most of these are implementation details of the concrete editor partials. I'd suggest only testing the attributes that actually appear in `_file.json.jbuilder`. Then there also isn't really a need to test all of the built in file types, since they are not a complete list of all possible file types anyway. 

What would be more interesting from my point of view would be to test that the specified `editor_partial` is actually included. Maybe the test could register a file type for `TestHostedFile` and provide an `editor_partial` [via `stub_template`](https://coderwall.com/p/fkh-fq/right-way-to-stub-views-in-helper-specs). That way we would actually test the file type API.